### ISSUE TITLE
NCSM - Posting Script Missing

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1683]
In GitLab by @kdjstudios on Apr 10, 2020, 11:35

Hello Team,

Cheryl just Messaged me saying that the NCSM site is missing it's upload option on the billing cycle. I am following up with her on what the site previously did for processing their billing. 

I do not recall the NCSM site having a posting script, but after looking at http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/524#note_56421 it would appear they did? If I am reading that comment correctly. But it looks like since we didn't have a switch name for it, I suspect it was removed when we released #524

May someone please provide more information on what that posting script did? 
*  If it was assigning usage to accounts then we will need to add it to the new universal posting program. I would suggest we name it "NCSM" with a version number "1.0.0"
* If it was not assigning usage to account, then I think the site should be good with using the No usage option, as nothing would be effected by the HR reports, invoice generation, etc. is that correct?

**EDIT:**

**Submitted by:** Cheryl Hamelin <cheryl.hamelin@answernet.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2020-04-10-64616/conversation

**Server:** Internal

**Client/Site:** NCSM

**Account:** NA

**Issue:** 

I am trying to create the NCSM 1st billing and it is not giving me the option to upload the dft file. Please help as this is critical.

We are already behind as we were waiting on some information.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 10, 2020, 12:00

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 10, 2020, 12:08

@amishra @ranand @smasih 

Please have the NCSM site working and able to process their billing ASAP.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 10, 2020, 12:25

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Apr 10, 2020, 12:38

@kdjstudios @Robleski @tim.traylor 

NCSM had no posting script on it before the release of #524, as mentioned in the given file in the [comment](http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/524#note_56240). 

So, after the release of #524 there is no switch on it. And upload usage file option/button will not be visible for the sites which don't have switch on them.

Sites without any switch need to click on the No Usage button to do their billing.  

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Apr 10, 2020, 13:45

@kdjstudios @Robleski @tim.traylor 

We are on the same team, we also want this issue to get resolved in the right manner. As we conveyed earlier that the NCSM site didn't have any switch, for the sites that didn't have the switch, we didn't create any posting program. 

For now to fix this, if we just show the upload button on the billing cycles page, it will work with its .sh file. So for now, we are doing that so that the site can do it's billing.

But we need a solution for it to work as per our new posting program.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Apr 10, 2020, 14:26

@kdjstudios @Robleski @tim.traylor 

We have fixed it on the production. We have made the upload button visible for the NCSM site(no switch) only. It will work with its .sh file.

But we need to find a way for it to work according to our posting program.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 10, 2020, 14:35

I wrote:

> The support team has just informed me there were able to update SA Billing and you should now be able to upload your usage file and proceed with processing the billing cycle. Please let us know if you have any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Apr 10, 2020, 15:27

@kdjstudios 

> May someone please provide more information on what that posting script did?

We looked at the .sh file. It is in PHP. It will take some time, to figure out what this file exactly does. We will figure it out tomorrow, then we can start the plan on adding it to our universal posting program.
 
Thanks

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 14, 2020, 01:03

assigned to @amishra

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 14, 2020, 03:22

@kdjstudios @tim.traylor 

We looked into this and our findings are described below:

| Site Name                           | Site Number | Usage File |
|-------------------------------------|-------------|------------|
| NCSM                                | 001         | YES        |
| Lowell,MA                           | 022         | No         |
| Billerica, MA (Outbound)            | 063         | No         |
| Webster City                        | 124         | No         |
| Energy Choice California            | 125         | No         |
| Hays, KS                            | 121         | No         |
| Bismarck, ND                        | 122         | No         |
| SA Hosted and Managed Services, SAB | SAB         | No         |
| Willow Grove, PA                    | 199         | No         |
| Fairlawn, OH                        | 070         | No         |
| Billerica, MA (Outbound)            | 063         | No         |


* Only NCSM site has PHP code inside the .sh file. Please find the PHP [file](/uploads/1220dacccbc0a51524c40af2d2684867/001.php)
* All the header fields defined inside the script file.
* When we upload the dft file they work the same as other posting scripts like VCC. like..get the details of each row in the dft file.
 
* The Rest of the site has only two line codes inside the .sh file.
`````ruby
#!/bin/bash
# copies file argument1 to file arugment2
cp $1 $2
`````
**Please let us know what we need to do further?**  

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 14, 2020, 11:25

@amishra @ranand @tim.traylor @Robleski 

Has this importing process been added to the universal posting script and HR?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 15, 2020, 24:46

@kdjstudios @tim.traylor 

Thank you for the update.

* As per the current functionality, there is no HR report and posting script(we have posting script but not like VCC ).
* In order to show the HR page, and posting script the same as other switches like VCC, we need to following data.
   * Human Readable Mapping
   * HR Report calculationXXX , code and calculation 
   * Usages File
   * Switch Name
 
 Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 15, 2020, 08:50

@amishra @tim.traylor @Robleski 

Thanks, lets discuss on Thursday on what to do. I believe everything you requested above we already have on the server.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 15, 2020, 22:40

@kdjstudios @tim.traylor 

Thank you for the update.
`````ruby
"9001" => "AGENT USAGE (PER MIN)",
"9002" => "CSM HOURS (PER HR)",
"9003" => "DEDICATED AGENT HOURS (PER HR)",
"9004" => "EXCESS SCHED FEE (PER INSTANCE)",
"9005" => "HIBERNATION FEE (FLAT)",
"9006" => "IVR USAGE (PER MIN)",
"9007" => "LANGUAGE USAGE (PER MIN)",
"9008" => "OUTBOUND AGENT USAGE (PER MIN)",
"9009" => "PATCHING FEE (PER CALL)",
"9010" => "PATCHING FEE (PER MIN)",
"9011" => "ROUTING FEE (PER MIN)",
"9012" => "SMS EXCESS (PER BLOCK)",
"9013" => "TIER 3 USAGE (PER MIN)",
"9014" => "TRAVEL FEES (FLAT)",
"9015" => "WEB PORTAL FEE (FLAT/ PER)",	
"9016" => "EXCESS TRANSCRIPTION",
"9017" => "CALLS DISPATCHED",
"9018" => "ACTIVATION USAGE",
"9019" => "LINE USAGE FEE",
"9020" => "OUTBOUND CALLING (PER HOUR)",
"9021" => "QUEUE USAGE (PER MIN)",
"9022" => "VRU USAGE (PER MIN)",
"9023" => "PROGRAMMING FEE (PER HOUR)",
"9024" => "TRAINING FEE (PER HOUR)",
"9025" => "REPORT FEE",
"9026" => "REMINDER CALLING (PER HOUR)",
"9027" => "CONFIRMATION CALLING (PER TRANS)",
`````

* Inside the .sh file we have only header & code.
* Rest of item we need to know like which switch name we need to setup, mock of the HR mapping etc..
* Also we have attached the .sh file on my above [comment](http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1786#note_57912). you can check as well.

Hope this helps.  

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 16, 2020, 09:38

1. Please use switch name "NCSM" and Version number "V1.0.0".
2. Please add an HR report with the above codes. Please use the layout above with three columns: 
   * "Name" This is the site's code name in SAB.
   * "Code" this is the billing code in SAB.
   * "Value" this is the imported value from the file.
   * "Usage File Header" this is the column header from the file. 
3. I will check with Cheryl if they can update the DFT file to look like the expected CSV file.
   * Will check with Cheryl on what the headers should be called.
4. The above will clarify how we add the importing process to the all posting program.

@ranand will provide the existing DFT file and the expected CSV version. To determine if operations can adjust the file type/format during their file generation.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Apr 17, 2020, 01:05

@kdjstudios @tim.traylor @Robleski 

As per yesterday's discussion, please find the attached NCSM usage file.

[dft_3.15.20.txt](/uploads/cda63272acb7995f9f311f9e4913d445/dft_3.15.20.txt)

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 17, 2020, 13:13

@ranand 

Thank you, I will review this with Cheryl and let you know what we are able to come up with for generating a CSV file instead.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 20, 2020, 23:14

@kdjstudios @tim.traylor 

Thank you for the update.

As per your last comment, we have updated the labels.

Please let us know what do we need to do.

thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 23, 2020, 13:55

@amishra @ranand @Robleski @tim.traylor 

I was able to talk with Cheryl and she was able to provide me with the original excel file and the CSV and DFT file outputs. 

[DFT_02.01.2020.csv](/uploads/9dfe4d807f24bf4cffecbcdff821a93c/DFT_02.01.2020.csv)

[DFT_02.01.2020.txt](/uploads/2662dd870e742d8a2eb15cfffb877a82/DFT_02.01.2020.txt)

[DFT_02.01.2020.xlsx](/uploads/261bb737138ea3d44903df6b1d726696/DFT_02.01.2020.xlsx)

I believe now what we need to do is add this as an importing process to the all posting program. I think the CSV file type is the best one correct? I do not believe there was a previous HR report for this, so that will need to be created too. I would expect this is still a priority item since this will effect their next Billing cycle.

Please provide an estimate first.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Apr 27, 2020, 04:13

As we will be converting the current .sh script to ruby script. Updating labels.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 27, 2020, 04:25

@kdjstudios @tim.traylor 

Thank you for the update

Please provide the HR MAPPING CODE page mockup.
Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 27, 2020, 08:10

@amishra @ranand  @Robleski @tim.traylor 

Here is the importing to the 7000 codes mapping. I believe we already have in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1786#note_57998 the mapping to the calculated codes.

[NCSM_Mapping_Document.csv](/uploads/92f97acb12543bcf0f1b9a85bbd925dc/NCSM_Mapping_Document.csv)

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Apr 27, 2020, 08:20

@kdjstudios @tim.traylor @Robleski 

@amishra  meant that we need the mock-up/ the page layout of the HR Report and hr mapping page. 

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 27, 2020, 10:23

@ranand @Robleski @tim.traylor 

Thanks please use the above table for the layout and mapping page. Just as with all other HR reports, please display the import values on the top half and the calculated codes and values on the bottom half.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Apr 28, 2020, 05:53

@kdjstudios @tim.traylor @Robleski 

Thank you for the update.

## Developer Estimate
|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|Medium-1|6|Worked with Concepts, implementation ok|8|Low Impact|3|11|

##### Estimate Notes:
* Risk areas identified: `Billing Process`

* **Work plan:**
  * Convert old script into the New Script.
  * Create new Switch & Switch Activities.
  * Create new HR Mapping Page Report
  * Testing

* We noticed that the **resouce_id** column is missing from the usage CSV file as you mentioned above in [comment](http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1786#note_58271)

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 28, 2020, 08:49

@amishra @tim.traylor @Robleski 

Which column/metric in the original posting script is SAB using to associate each row with an account? That should be the same column we use now, I would expect. NOTE: I checked the Exceptions report and there are no accounts listed there for the NCSM site, so it would only make sense that one of the columns we are using to match. Maybe the Account Number column?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 30, 2020, 08:59

This was reviewed today and @ranand stated that we are using the account column to identify the accounts currently.

**Comment by kdjstudios** [comment]
In GitLab by @gregory.robleski on Apr 30, 2020, 09:00

i agree and approve

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on May 5, 2020, 07:04

@kdjstudios  @tim.traylor  @Robleski 

Thanks for the update.

We have completed it's development and setup branch on the merge2 server for your testing. 


![image](/uploads/fa9f7628989ab544f914545ae3d93d12/image.png)

We have the following query:

* What we need to show on the code mapping page for a **Resource ID**? As its hardcoded field.

* We ware fetching account with **resource id** for **VCC Switch** by prepending `0` on resource id. Do we need to do the same for the account number?

* There is no posting factor in the .sh script. So we have set the posting factor to 1. Is it ok or do we need to set it to something else?

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 5, 2020, 12:23

@rranjan @Robleski 

* What we need to show on the code mapping page for a **Resource ID**? As its hardcoded field.
> The resource ID needs to be set as it was previously. I believe when this was reviewed in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1786#note_58556 it was set from the Account Number column. Is that correct?

* We ware fetching account with **resource id** for **VCC Switch** by prepending `0` on resource id. Do we need to do the same for the account number?
> Yes please do the same prepending of the "0" if no exact match is found the first time, just like in the VCC. 

* There is no posting factor in the .sh script. So we have set the posting factor to 1. Is it ok or do we need to set it to something else?
> Yes please ass the posting factor and set it to 1. 

Please let us know if there are any other questions

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 5, 2020, 12:28

I received the following error when attempting to upload the usage file from http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1786#note_58271. I presume that is because of the above responses were not answered so this was actually not ready for testing. Please let me know once this has been tested and is ready for us to test. 

![image](/uploads/d04f9d002d97ca2242b7d7cafe159221/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on May 6, 2020, 01:54

@kdjstudios  @tim.traylor  @Robleski 

Thanks for the update.

We looked into it and following is our finding:- 

*  Usage file given on comment http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1786#note_582 had missing header **Account Name** So please use [DFT_02.01.2020__2___1_.csv](/uploads/010bd7084aa0d242e6dd0a019e9feca3/DFT_02.01.2020__2___1_.csv)

*  We also noticed that you did upload usage for the terminated account. So in that case, we are not doing anything. 

We have again set up a branch on the merge2 server for your testing.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @gregory.robleski on May 6, 2020, 10:36

kyle to retest

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 6, 2020, 13:59

@rranjan @Robleski 

I was able to retest this, and everything looks good functionality wise. But it looks like at least in the HR report we are not using the assigned 7000 codes in the mapping section, instead we are directly assigning the the 9000 values. Which also is then incorrect in the Calculated codes section too. Please correct this.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on May 6, 2020, 20:33

@kdjstudios  @Robleski 

Thanks for the update

We have fixed the hr report and setup branch on the merge2 server for your testing.


Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 7, 2020, 08:32

@rranjan @amishra @ranand @Robleski @tim.traylor 

Okay still have issues with the codes. The billing codes setup at the site level are the codes that should be displayed in the "Codes" column. ALso the Name column needs to display what is setup at the site level too. The 7000 codes assigned by the posting program should be displayed in the Mapping document and in the "Calculation with 7XXX codes" column. This has been how the HR report works for every site, Why are we trying to do things differently now?

![image](/uploads/eca5406071759899eeac025526c83922/image.png)

I am still looking into the exception report, but please confirm that it working for this site too.

I was able to download the usage file ([dft_02.01.2020__2___1_.csv.20200506225614.20200506225614.csv](/uploads/0b4387da74fc08994512c6dae02f2f47/dft_02.01.2020__2___1_.csv.20200506225614.20200506225614.csv)) from SAB and it appears to have a few accounts listed as "Not Found". However I do not recall seeing these listed on the "Missing accounts" page, nor do I see them on the exceptions report. May someone please look into this and confirm why it is not working as expected.

**Comment by kdjstudios** [comment]
In GitLab by @gregory.robleski on May 7, 2020, 08:49

needs recoding + retesting. may not make the 5/9 SAB release

**Comment by kdjstudios** [comment]
In GitLab by @ranand on May 7, 2020, 09:13

@kdjstudios @Robleski 

The mapping file given to us in the http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1786#note_58398 has all 7000 codes. And the mapping given by Ashish had 9000 codes in the http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1786#note_57998. 
**which is the from them is correct mapping of codes?** 

NOTE: As far as I know, NCSM is using all 9000 codes only. And .sh posting script was also based on the 9000 codes.

Thanks.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 7, 2020, 09:35

@ranand @amishra @Robleski 

## HR
As we have always done on all HR reports:

In the mapping document, we assign the 7000 import codes. 

In the Calculated fields section:
1. Name - this is the name from the site level
2. Code - this is the code from the site level
3. Value - This is the metric result/value.
4. Usage File headers - This is the formula using the headers of the columns from the usage file.
5. 7XXX codes - This is the formula using the 7000 codes from the mapping document.


## Missing Account / Exceptions Report

As mentioned in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1786#note_58742 the missing accounts and exceptions report do not appear to be working.

**Comment by kdjstudios** [comment]
In GitLab by @gregory.robleski on May 8, 2020, 07:02

expected to be released for AN testing this am.

**Comment by kdjstudios** [comment]
In GitLab by @rranjan on May 8, 2020, 09:04

@kdjstudios  @tim.traylor  


Thank you 

We have fixed it and set up a branch on the dev server for your testing.

> I am still looking into the exception report, but please confirm that it working for this site too.

 We looked into it but the exception report was not working for this site in the past.

> I was able to download the usage file ([dft_02.01.2020__2___1_.csv.20200506225614.20200506225614.csv](http://gitlab.aavaz.biz/AnswerNet/SABilling/uploads/0b4387da74fc08994512c6dae02f2f47/dft_02.01.2020__2___1_.csv.20200506225614.20200506225614.csv)) from SAB and it appears to have a few accounts listed as "Not Found". However I do not recall seeing these listed on the "Missing accounts" page, nor do I see them on the exceptions report. May someone please look into this and confirm why it is not working as expected.

The missing account page will show only when the account number in the usage file is invalid. The given usage file has all the valid accounts.

NOTE: It does not show exception for inactive accounts nor it does upload the usage for the inactive account. It is currently working exactly like .sh script.
 
Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 8, 2020, 12:05

@rranjan @ranand @Robleski 

Here is the file I was using to test, which has s "Test" account that is not in SAB. The account appeared on the "Missing accounts" page, and also appeared on the exceptions report. However when i take a look at the file I can download from SAB, it appears there are lots of accounts listed as "Account Not Found" may someone please explain to me why they are listed as not found, but did not appear in the missing accounts or exceptions.  

Uploaded:[DFT_02.01.2020__2___1__Test.csv](/uploads/4b75650ba359a4b3c267c254074fdcc6/DFT_02.01.2020__2___1__Test.csv)

Download: [dft_02.01.2020__2___1__test.csv.20200508135934.20200508135934.csv](/uploads/3e9eb40cefc04bfd8d3e3efadbf1f8b1/dft_02.01.2020__2___1__test.csv.20200508135934.20200508135934.csv)

Other then that, this appears to be functional. so marking as ready to be released.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on May 9, 2020, 24:47

@kdjstudios @Robleski 

* Accounts that have status as "Account Not Found" are all inactive Accounts.

* SAB prepends "T-" to the account numbers of the inactive accounts.

* Since the uploaded usage file doesn't have "resource ID" column in it, the script creates a duplicate column of "Account Number" with the heading of "Resource ID"

* When the script creates the downloadable/output CSV, it looks for the account through "Account Number". And it doesn't find these inactive accounts, as the usage file doesn't have "T-" prepended to the account number, so it adds the status as "Account Not Found".

* The Exception report/ missing account report gets created from the SAB code not from the script. While creating the exception report it searches the accounts through the "resource_id". It finds all inactive accounts by their resource id, so they don't show up on the exception report.

NOTE: Given usage file's account number is copied as resource id. For inactive accounts we only prepend "T-" to their account number and not to their resource id. It is working as it was working with .sh file.

Please notify the site operations that from now on they need to upload CSV format of the usage and with the exact same headers

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @amishra on May 9, 2020, 03:40

@kdjstudios @tim.traylor @Robleski 

In addition to @ranand comment:

We have successfully released on the production server.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 9, 2020, 13:17

@ranand @amishra 

Thank you for the update and clarifications. I think we will need to review this with @Robleski on Monday. Since it appears we are using different methods to confirm if an account exists. which we should not be. It should always be the same. 

NOTE: I would presume we will want to ensure we are always using the resource ID from SAB when comparing to an uploaded file. IDK where/when it was requested to use the SAB account number. I know in the usage file it is labeled as "account number" but that is because it does not come from SAB, so we can not control the header names.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 9, 2020, 13:21

@Robleski @amishra @ranand 

After testing on Production. it appears that the Calculation columns are not populating in the HR. Who tested this and how did you think this would pass testing? Please fix this before Monday morning. 

![image](/uploads/f09bb55819356770fdcf3d0cec65b17e/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @ranand on May 11, 2020, 05:41

@kdjstudios @Robleski 

* **The given HR-Report is the report previous to the release of this switch.**
* The released switch start date is its release date ie. 09-May-2020. All the reports prior to that will have activities and their calculations come from the account. 

We couldn't test the NCSM switch upload process on the production server. So, for testing we followed all the steps of release on the dev server and did our testing on it before it's release to the production server. 

* To fix the historical HR report we have 2 options, either we update the switch start date to the site creation date or we update the calculation header and calculation 7XXX codes of the account activities. Please confirm, which process we should follow?

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @gregory.robleski on May 11, 2020, 07:06

greg to follow up with kyle and confirm ; avaaz will updating update the switch start date to the site creation date.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on May 11, 2020, 08:03

@Robleski 

As discussed, we have updated the switch's start date to the site creation date.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 11, 2020, 08:33

@ranand 

Thank you, This is now working as expected. Please remember if you do not know what to do due to it not being specified or if there is confusion, please ASK; do NOT just make your own decisions on how SAB should work. Thanks @Robleski for clarifying to use the site's start date.

**Comment by kdjstudios** [comment]
In GitLab by @gregory.robleski on May 11, 2020, 08:37

so based upon the threads, this ticket is in prod and is working properly?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on May 13, 2020, 09:52

@Robleski 

It is in Prod, but we will need to wait until the site runs their next billing cycle to fully confirm it is working as expected. However, looking at past billing cycle's the HR report part of this update appears to be working successfully, but we still need to know that processing the billing cycle steps work as expected. I have already reached out to Cheryl and we plan on doing that together just to ensure they do not encounter any errors. I will post an update once we can confirm to close.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 9, 2020, 10:40

@amishra @ranand @rranjan @Robleski 

Here is the original file [Copy_of_DFT_06.01.2020.xlsx](/uploads/daadb81b82420ccfe227d7369338323d/Copy_of_DFT_06.01.2020.xlsx). We exported to CSV [CSV_UTF_DFT_06.01.2020.csv](/uploads/376804fcfa6ca45557b4605bb42989a5/CSV_UTF_DFT_06.01.2020.csv) and all accounts are showing as missing. I downloaded it here [dft_06.01.2020_csv.csv.20200609121628.20200609121628.csv](/uploads/8e26750caf554df0540fe0f5ef939a27/dft_06.01.2020_csv.csv.20200609121628.20200609121628.csv). All accounts are not being found. What is going on?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 9, 2020, 10:48

Team, While testing with this I noticed that the column B header "Account Name" has been missing on the latest spreadsheets. I manually added that to the file and re-uploaded. here is the download from the uploaded file: [dft_06.01.2020_modified.csv.20200609124154.20200609124154.csv](/uploads/33e1ffaf8b5b39cba6de0f9b1120a513/dft_06.01.2020_modified.csv.20200609124154.20200609124154.csv) This time it is showing the account Name on the missing accounts page, but that is also missing it's header there too (Highlighted Red).

It seems to be that SAB is not able to find the accounts? What happend this was working when we tested it? is there something that ops changed on the file or is SAB not matching the Account Number column from the file to the SAB Resource ID (highlighted Blue).


![image](/uploads/acfd015774d1c454d8e2f45f054a01f2/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jun 9, 2020, 11:18

File has a blank header, which should be "Account Name".

I've updated the usage header

[CSV_UTF_DFT_06.01.2020.csv](/uploads/6651de174a0a5298435144a8e3a54085/CSV_UTF_DFT_06.01.2020.csv)

I've uploaded the same usage file on the DEV server. And it is showing 5 missing accounts now.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 9, 2020, 11:31

@ranand 

Please read: http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1786#note_59758

**UPDATE:**

I just took the orginal file from: http://gitlab.aavaz.biz/AnswerNet/SABilling/uploads/daadb81b82420ccfe227d7369338323d/Copy_of_DFT_06.01.2020.xlsx and added the missing header. I then saved as CSV [CSV_Header_DFT_06.01.2020.csv](/uploads/6d69736c707c789bb4407342cf94ab85/CSV_Header_DFT_06.01.2020.csv) and tried to upload to the 1st Billing cycle on the NCSM site on the DEV server. I got the following error:

![image](/uploads/1bc05ee8631485079e2fe3d08cafde65/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 9, 2020, 11:35

Just found another concern:

Take example the first account on the usage that is uploaded: 001-0020	Patriot Communications LLC. This account is setup under the 1st Billing cycle, not the Master. So why is it not showing under the missing accounts on the master, and why is it showing in the missing accounts on the 1st Billing. Something is not right. 

![image](/uploads/7bb9f2a30d548af2aa996336e1249f67/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 9, 2020, 11:40

I also just saw this error come in via email. Possible another issue?

![image](/uploads/7e6047e099111fb669c7e055b06a1aac/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jun 9, 2020, 11:40

It was on the dev server. Dev server had another dev branch set on it

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jun 9, 2020, 11:54

I uploaded this file on the dev server and it is working fine.
[CSV_Header_DFT_06.01.2020__1_.csv](/uploads/c99c79f3edb38d5e5573c5b51a700349/CSV_Header_DFT_06.01.2020__1_.csv)

Can you please try again?

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 9, 2020, 12:59

I was able to process the Billing cycle to the apply step for the NCSM site and walk them through the processes. Below are the items that we need to ensure are working by the 15th as that is when they runt heir next billing cycle. 

### Import/Upload Process:
1. Original file attached: [Copy_of_DFT_06.01.2020.xlsx](/uploads/fa56e09c3d4d9cd8157d21f04f221220/Copy_of_DFT_06.01.2020.xlsx). DO NOT MODIFY THIS IN ANY WAY. Only save as "CSV UFT-8".
2. SAB needs to be able to import the CSV file that was exported from step 1 without any modifications. (Attached: [Copy_of_DFT_06.01.2020.csv](/uploads/42c887bfa40a7ea27cfe10ec39cc8217/Copy_of_DFT_06.01.2020.csv)
3. Insure they import correctly with the correct Billing Cycle accounts.

### Missing accounts page:
1. Add the "Account Name" Header - screen shot in previous post http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1786#note_59758

### Exceptions report:
1. This looks good.

### HR Report:
1. This looks good.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 10, 2020, 06:37

@kdjstudios  @Robleski 

Thank you for the update.

> Original file attached: [Copy_of_DFT_06.01.2020.xlsx](http://gitlab.aavaz.biz/AnswerNet/SABilling/uploads/fa56e09c3d4d9cd8157d21f04f221220/Copy_of_DFT_06.01.2020.xlsx). DO NOT MODIFY THIS IN ANY WAY. Only save as "CSV UFT-8".

* I would like to inform you that we followed the same steps and successfully upload the file.   
Steps we took:    
  * We downloaded the original file.   
  * Converted it into the CSV UTF-8.   
  * And then uploaded it   
We have attached a video of our step. Please have a look.

https://www.loom.com/share/2c6792096e4f45cf827c4bdcedeea196

* Please let us know if we missed something.

> SAB needs to be able to import the CSV file that was exported from step 1 without any modifications. (Attached: [Copy_of_DFT_06.01.2020.csv](http://gitlab.aavaz.biz/AnswerNet/SABilling/uploads/42c887bfa40a7ea27cfe10ec39cc8217/Copy_of_DFT_06.01.2020.csv)

* We tried to find differences in both files but did not find anything yet.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 10, 2020, 08:48

@amishra @Robleski 

The main difference I see is that you are exporting not using MS Excel. There has to be something different with the files if one file is able to be uploaded and not show every account as missing; and the other file it shows all accounts are shown as missing? Do you need to do like a "git dif" type file comparison or is there some other type of file comparison app that can see the differences.

Also I saw the same error as in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1786#note_59762 a few times, I presume that occured when you were uploaded the Excel CSV file, is there something with the formating on line 211?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 11, 2020, 05:00

@kdjstudios @Robleski 

Thank you for the update.

* Yes, Because we are using Ubantu OS and Ubantu has by default libre Office.
* But we had converted with UTF-8 and .csv format as you can see on the above-mentioned video.

* We would like to inform you that we have checked with online file diff tool and found only decimal points differences.

* **IT WOULD BE GREAT IF YOU CAN POST A SORT VIDEO WHICH STEP YOU ARE PERFORMING.**

thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 11, 2020, 08:50

This was reviewed on the call today. The Aavaz team will use the files we have provided to ensure SAB is able to upload correctly.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jun 12, 2020, 07:15

@kdjstudios @Robleski 

After much digging, we found that the Excel file you were saving was creating a UTF-8 BOM file. BOM stands for "Byte Order Mark". 

Your file also had an extra comma at the end of the file, as well as 4 lines of blank data at the end of the file. 

The other differences between yours and ours were decimal places on the data, and CRLF vs LF. 

![Image_Pasted_at_2020-6-12_12-28](/uploads/e2040b8c2a35684e89927bd328358cc2/Image_Pasted_at_2020-6-12_12-28.png)

The main issue was the BOM characters. 

If we strip those then everything works fine. 

We will need to change the existing NCSM posting script to handle both BOM and non-BOM files. It should take **4 to 5 hours** to get it done. That way different versions of Excel to CSV won't cause this problem. 

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @gregory.robleski on Jun 12, 2020, 09:40

ok approved to start work

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 15, 2020, 07:48

@kdjstudios @Robleski 

Thank you for your approval.

We have successfully completed the development and step branch for your testing on the dev server.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 15, 2020, 09:28

This now appears to be working. We will need to push this to production over night, as the NCSM site has multiple billing cycles and I believe they have one that started on the 12th (Master)

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jun 16, 2020, 01:22

@kdjstudios @Robleski 

Thank you for the update.

We have successfully released it on the production server.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 16, 2020, 08:16

@amishra 

Thank you, I will confirm with Cheryl that this is working for ops.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 23, 2020, 15:11

closed
